_posts/en/newsletters/2018-11-13-newsletter.md

```diff
@@ -0,0 +1,141 @@
+  order to allow development to focus on the project's main objectives.
+
+    This does provide a nice opportunity for an independent developer or
+    other third-party to create a separate project for a library, local
+    program, or RPC interface that provides a stable interface to
+    utility functions that work well in conjunction with Bitcoin Core,
+    and which perhaps even provides some of the utility functions that
+    Bitcoin Core already supports---giving the Bitcoin Core project the
+    ability to gracefully deprecate those options and recommend the new
+    tool to users needing those functions.  Some ideas for how to
+    implement such a tool were discussed both during and [after][core
+    dev log] the meeting.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][core commits],
+[LND][lnd commits], [C-lightning][cl commits], and [libsecp256k1][secp
+commits].*
+
+{% include linkers/github-log.md
+  refname="core commits"
+  repo="bitcoin/bitcoin"
+  start="742ee213499194f97e59dae4971f1474ae7d57ad"
+  end="e70a19e7132dac91b7948fcbfac086f86fec3d88"
+%}
+{% include linkers/github-log.md
+  refname="lnd commits"
+  repo="lightningnetwork/lnd"
+  start="6b19df162a161079ab794162b45e8f4c7bb8beec"
+  end="d4b042dc1946ece8b60d538ade8e912f035612fe"
+%}
+{% include linkers/github-log.md
+  refname="cl commits"
+  repo="ElementsProject/lightning"
+  start="d5bb536ef0c08a813f767b3fb016eb20292de4dd"
+  end="62e6a9ff542e40364b67a7aa419e33ed72b96a42"
+%}
+{% include linkers/github-log.md
+  refname="secp commits"
+  repo="bitcoin-core/secp256k1"
+  start="1086fda4c1975d0cad8d3cad96794a64ec12dca4"
+  end="1086fda4c1975d0cad8d3cad96794a64ec12dca4"
+%}
+
+- [Bitcoin Core #14410][] adds an `ischange` field to the
+  [getaddressinfo][rpc getaddressinfo] RPC indicating whether the wallet
+  used the address in a change output.
+
+- [Bitcoin Core #14060][] makes configurable the maximum number of
+  messages the [ZeroMQ][] (ZMQ) interface will queue for a client.  The
+  default High-Water Mark (HWM) allows up to 1,000 messages to be queued
+  before some messages are dropped.  A new HWM may be chosen by setting
+  one of the following configuration options to the desired maximum
+  number of queued messages (or the maximum queue size may be made
+  unlimited by setting it to `0`): `zmqpubhashtxhwm`,
+  `zmqpubhashblockhwm`, `zmqpubrawblockhwm`, and `zmqpubrawtxhwm`.
+  The greater the queue size, the more memory the program will use.
+
+- [LND #1782][] adds a `num_inactive_channels` field to the `getinfo` RPC
+  with the number of the node's inactive channels (similar to the
+  existing counts of pending and active channels).
+
+- [LND #1944][] adds a `pub_key` field to the `sendtoroute` RPC so that
+  LND doesn't need to get the pubkey from an external source.  This
+  allows routing payments through private channels that are not listed
+  on the public network.
```

Comment:
I can't think of a reason why that'd be a useful feature.  You can do remote authorization (e.g. channel on server, keys on mobile) without creating a separate channel requiring occasional on-chain txes between the server and mobile.  What am I missing?

Reply:
My (mostly uninformed) understanding of lightning routing is that keeping an up-to-date graph of channels could be quite resource intensive, and therefore difficult to maintain on devices with limited compute or bandwidth. Allowing the route to be specified in `sendtoroute` would allow a node to run on those devices without maintaining that expensive channel graph.

Reply:
This is a secondary motivation for the PR, so I think it's fine to leave it out. If there are further PRs to allow graphless nodes then we can cover those in the newsletter.